java-frontend/src/main/java/org/sonar/java/se/FlowComputation.java

```diff
@@ -51,191 +51,211 @@
 
   private final Predicate<Constraint> addToFlow;
   private final Predicate<Constraint> terminateTraversal;
-  private final List<JavaFileScannerContext.Location> flow = new ArrayList<>();
-  private final SymbolicValue symbolicValue;
-  private final Set<ExplodedGraph.Node> visited = new HashSet<>();
+  private final Set<SymbolicValue> symbolicValues;
   private final List<Class<? extends Constraint>> domains;
 
   private FlowComputation(@Nullable SymbolicValue symbolicValue, Predicate<Constraint> addToFlow,
                           Predicate<Constraint> terminateTraversal, List<Class<? extends Constraint>> domains) {
     this.addToFlow = addToFlow;
     this.terminateTraversal = terminateTraversal;
-    this.symbolicValue = symbolicValue;
+    this.symbolicValues = computedFrom(symbolicValue);
     this.domains = domains;
   }
 
-  public static List<JavaFileScannerContext.Location> flow(ExplodedGraph.Node currentNode, @Nullable SymbolicValue currentVal, List<Class<? extends Constraint>> domains) {
+  private static Set<SymbolicValue> computedFrom(@Nullable SymbolicValue symbolicValue) {
+    if (symbolicValue == null) {
+      return Collections.emptySet();
+    }
+    Set<SymbolicValue> result = new HashSet<>();
+    result.add(symbolicValue);
+    symbolicValue.computedFrom().forEach(sv -> result.addAll(computedFrom(sv)));
+    return result;
+  }
+
+  public static Set<List<JavaFileScannerContext.Location>> flow(ExplodedGraph.Node currentNode, @Nullable SymbolicValue currentVal, List<Class<? extends Constraint>> domains) {
     return flow(currentNode, currentVal, constraint -> true, domains);
   }
 
-  public static List<JavaFileScannerContext.Location> flow(ExplodedGraph.Node currentNode, @Nullable SymbolicValue currentVal, Predicate<Constraint> addToFlow,
+  public static Set<List<JavaFileScannerContext.Location>> flow(ExplodedGraph.Node currentNode, @Nullable SymbolicValue currentVal, Predicate<Constraint> addToFlow,
                                                            List<Class<? extends Constraint>> domains) {
     return flow(currentNode, currentVal, addToFlow, c -> false, domains);
   }
 
-  public static List<JavaFileScannerContext.Location> flow(ExplodedGraph.Node currentNode, @Nullable SymbolicValue currentVal, Predicate<Constraint> addToFlow,
+  public static Set<List<JavaFileScannerContext.Location>> flow(ExplodedGraph.Node currentNode, @Nullable SymbolicValue currentVal, Predicate<Constraint> addToFlow,
                                                            Predicate<Constraint> terminateTraversal, List<Class<? extends Constraint>> domains) {
     FlowComputation flowComputation = new FlowComputation(currentVal, addToFlow, terminateTraversal, domains);
-
+    // FIXME SONARJAVA-2049 getLastEvaluated doesn't work with relational SV
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1134)